src/components/InfluenceChart/index.js

```diff
@@ -0,0 +1,751 @@
+  const currentIds = union(new Set([focus.id]), influencedBy, influenced)
+  const currentPeople = new Set(fp.compose(
+    fp.filter(p => p != null),
+    fp.map(id => people[id]),
+  )(Array.from(currentIds.values())))
+  const oldPeople = new Set(fp.map(n => n.person)(graph.getVisibleNodes()))
+
+  const incomingPeople = difference(currentPeople, oldPeople)
+  const outgoingPeople = difference(oldPeople, currentPeople)
+
+  graph.setFocus(focus)
+  incomingPeople.forEach((p) => {
+    if (p === focus) {
+      return
+    }
+    graph.addPerson(p)
+    if (influencedBy.has(p.id)) {
+      graph.createLink(p, focus)
+    } else {
+      graph.createLink(focus, p)
+    }
+  })
+
+  outgoingPeople.forEach((p) => {
+    graph.removePerson(p)
+  })
+}
+
+
+class InfluenceCanvas {
+  focus: PersonDetail
+  people: PeopleCache
+  timeline: Timeline
+
+  dimensions: Dimensions
+  graph: TGraph
+
+  topElem: Selection
+  definitions: Selection
+  nodesElem: Selection
+  linksElem: Selection
+  lifelinesElem: Selection
+
+  timelineAxis: Selection
+  fdl: ForceSimulation
+  fdlLinks: LinkForces
+
+  selectNode: (PersonAbstract | PersonDetail) => void
+
+  highlight: ?PersonNode
+
+  constructor(
+    topElem: Selection,
+    dimensions: Dimensions,
+    focus: PersonDetail,
+    people: PeopleCache,
+    selectNode: (PersonAbstract | PersonDetail) => void,
+  ) {
+    this.topElem = topElem
+    this.dimensions = dimensions
+    this.focus = focus
+    this.people = people
+    this.graph = new TGraph(focus)
+    this.selectNode = selectNode
+
+    // create clip path for image
+    this.definitions = this.topElem.append('defs')
+
+    this.definitions.append('svg:clipPath')
+      .attr('id', 'image-clip')
+      .append('svg:circle')
+      .attr('cx', 0)
+      .attr('cy', 0)
+      .attr('r', IMAGE_SIZE / 2)
+
+    /* I've put these here so that I can force them to be rendered in a
+     * particular order. If all of the links appear in one container, and all
+     * of the nodes appear in another container, and the nodes container comes
+     * after the links container, this makes the nodes render atop the links.
+     */
+    this.lifelinesElem = this.topElem.append('g').classed('timelines', true)
+    this.linksElem = this.topElem.append('g').classed('links', true)
+    this.nodesElem = this.topElem.append('g').classed('nodes', true)
+
+    const [minYear, maxYear] = calculateTimeRange(listOfPeopleInGraph(this.graph, this.people))
+    this.timeline = createTimeline(this.dimensions.width, minYear, maxYear)
+    this.timelineAxis = topElem
+      .append('g')
+      .classed('axies', true)
+      .attr('class', 'axis')
+      .attr('transform', `translate(0, ${TIMELINE_Y(dimensions.height)})`)
+      .call(this.timeline.axis)
+
+    this.fdl = d3.forceSimulation()
+    this.fdlLinks = d3.forceLink()
+      .strength(LINK_STRENGTH)
+      .distance(() => (Math.random() * LINK_RANDOM) + ((NODE_SIZE / 2) + LINK_MIN_OFFSET))
```

Comment:
What is NODE_SIZE as opposed to IMAGE_SIZE? Are they the same, and should they be?

Reply:
They are different, and I see that they represent nuanced differences in now a node gets displayed vs. how links to it are calculated. Those differences should probably be handled programmatically, instead of hard-coded.